src/ufuncs.c

```diff
@@ -767,7 +1003,7 @@ static char snap_dtypes[4] = {NPY_OBJECT, NPY_OBJECT, NPY_DOUBLE, NPY_OBJECT};
 static void snap_func(char **args, npy_intp *dimensions,
```

Comment:
Should this be updated to use `GEOS_INIT_THREADS` / `GEOS_FINISH_THREADS`?  If not, it might be good to add a comment indicating why it is not compatible with multiple threads.

Reply:
Thanks for spotting that omission, I added it. Also, the creation functions like `points` need attention still, but I'd like to leave that for later.

Care to take a look at the new `snap_func`?